### PR TITLE
[Qwen3-TTS] Stateful incremental codec decoding by default: worker streams, arena-bound graphs, event-ordered decode

### PR DESCRIPTION
**Pull request by luojiaxuan**
## Summary

Ships the stateful incremental codec decoder for streamed Qwen3-TTS by default and fixes what kept it from beating the windowed decoder. Builds on #1846 (@BruceLoveDecimal, arena-backed incremental state) and #1855 (@leihehehe, CUDA graphs for the incremental step), both merged in here with their review findings addressed. Tracking: #1754.

What changed on top of those two PRs:

- Incremental cohorts decode on the worker's own stream; WARM graphs are captured per follow-up worker and COLD (bootstrap) graphs per batch bucket, so a graph exec is never shared across streams.
- The decode step is a tensor-only kernel with `torch.compile` for the steady WARM shapes (1147 kernels to 356, replay 3.2 ms to 1.8 ms).
- Graphs gather from and scatter to the arena inside the replay, so no per-cohort host copies of the state.
- One cohort stays in flight per follow-up worker (launch-ahead), with a bounded wait on the shared collect lock so a worker that owes a commit never waits for unrelated traffic.
- Decode workers no longer touch the default stream: the talker records an event when a chunk is ready, the vocoder waits on that event, zeroes arena slots and screens codes on its own stream, and keeps the chunks a plan reads alive until that plan is committed. The same for in-process preprocessing. Before this every vocoder decode queued behind the whole in-flight talker step.
- Arena index staging goes through a ring of pinned buffers (two cohorts can be in flight per worker).

Defaults: `enable_stateful_codec_decoder=True`, `followup_batch_wait_ms=4`; the CUDA graph and compile switches follow the stateful decoder. `enable_stateful_codec_decoder=False` restores the windowed decoder.

## Results

eval-h100 (one H100 80GB), CustomVoice 1.7B, seed-tts-eval prompts, open-loop Poisson arrivals, three client seeds per arm, 100% completion on every run. Baseline is the same tree with `enable_stateful_codec_decoder=False`, i.e. the windowed decoder with the earlier chunk-ramp and follow-up-window work.

| Metric | windowed decoder | this PR |
|---|---|---|
| Underrun, rps 20, default ramp (1,2,4) | 20.9% | 0.6-2.3% (9 seeds, mean ~1.2%) |
| Underrun, rps 20, ramp (2,4) | | 0-0.5% |
| First playable p50, rps 20 | 82-89 ms | 55-58 ms |
| First playable p50, rps 1 | 44.5 ms | 35.6 ms |
| Audible TTFA p50, rps 1 | | 44 ms |

Single-seed noise on underrun is about one percentage point at rps 20. Graph memory is 0.50 GB.

## Validation

- `tests/unit_test/qwen3_tts/` (507 passed; the three `test_compat` docs tests fail the same way on main in this environment).
- rps 1 acceptance with 47/47 requests and zero underrun on the default tree and on the eager-incremental control; this is where the launch-ahead stall and the stream coupling were found, neither is visible at rps 20.
- Real-code numerical check of the incremental path against the windowed decoder (rtol 2e-4).

Negative results kept out of the defaults: `initial_batch_wait_ms=0` (worse underrun), talker mixed chunked prefill (collapses at rps 20), emitting the silent bootstrap frame (TTFB 27.7 ms but audible TTFA 107 ms).

### CODE REVIEW

**Comment by luojiaxuan** [comment]
Self-review landed in 4ccf87ec and 4f47debf, net 358 lines removed. Fixes worth a look:

- A follow-up batch that mixed an incremental cohort with left-context streams could hand the eager decode a pinned slot still owned by an in-flight cohort, failing those streams; legacy-only batches also never drained the pending cohort. Both fixed by finishing in-flight cohorts before the legacy loop.
- The ref-prefixed first chunk is built by a `torch.cat` after the step's ready event, so voice-clone requests now get their own event after the cat.
- Arena slots are released with a CUDA event and the next owner waits on it before zeroing.
- One launch/finish path for cohorts; rows rejected before decode fail individually instead of dropping the whole cohort to the legacy decoder.
- Runner `stats()` reads under a lock; the non-arena graph path, its result type and the duplicated teardown are gone.
- Plans keep their chunks alive until commit instead of calling `record_stream` per chunk under the state lock.

Unit tests pass (514 in `tests/unit_test/qwen3_tts/`). No default changed.

File: docs/cookbook/qwen3_tts.md [hunk 1]
```diff
@@ -387,6 +387,28 @@ explicit value to trade continuity against time-to-first-audio.
...
+  vocoder:
+    factory:
+      enable_stateful_codec_decoder: false
+```
+
+`incremental_codec_cuda_graph`, `incremental_codec_compile` and
+`followup_batch_wait_ms` are the individual switches. At 20 requests per
+second on one H200 the default path holds 2.7% of streams underrun against
+5.4% for the left-context decoder at the same first-audio latency; the
+measurements are in `docs/benchmarks/qwen3_tts_r20_vocoder_study.md`.
```
Comment (JiaxinD) [inline docs/cookbook/qwen3_tts.md]: `docs/benchmarks/qwen3_tts_r20_vocoder_study.md` is not in the tree (`docs/benchmarks/` holds only `relay.md`), and this paragraph reports one H200 at 2.7% against 5.4% while the PR body reports eval-h100 at 20.9% against 0.6 to 2.3% for the same metric and rps.
One of the two is stale.

**Review by JiaxinD — COMMENTED** [review]
Note: (the body still says chunks are pinned with `record_stream`, but 4f47debf replaced that with retained references, and `record_stream` now only appears in `request_builders.py`)

File: docs/cookbook/qwen3_tts.md (hunk 1 above)
Comment (luojiaxuan) [reply docs/cookbook/qwen3_tts.md]: Both stale, fixed.

The cookbook paragraph now reports the H100 80GB measurements this PR was validated on (0.6% to 2.3% underrun against 20.9%, first playable 55 to 58 ms against 82 to 89 ms) and drops the reference to the benchmark doc, which lives on my run-doc branch and is not in the tree. The H200 numbers were from the earlier round before the stream decoupling landed. Fixed in c53caab3.

PR body updated too: 4f47debf replaced the per-chunk `record_stream` with plans holding their chunks until commit, so `record_stream` now appears only in `request_builders.py`.

**Review by JiaxinD — APPROVED** [review]
All three fixed: the cookbook carries the validated H100 numbers, the dangling benchmark reference is gone, and the body now describes the chunk keepalive as implemented.
